src/components/metadataSelects/CategoryComboSelect/CategoryComboSelect.tsx

```diff
@@ -0,0 +1,122 @@
+import React, { useCallback, useRef, useState } from 'react'
+import { SelectOption } from '../../../types'
+import { SearchableSingleSelect } from '../../SearchableSingleSelect'
+import { useInitialOptionQuery } from './useInitialOptionQuery'
+import { useOptionsQuery } from './useOptionsQuery'
+
+function computeDisplayOptions({
+    selected,
+    selectedOption,
+    options,
+}: {
+    options: SelectOption[]
+    selected?: string
+    selectedOption?: SelectOption
+}): SelectOption[] {
+    // This happens only when we haven't fetched the lable for an initially
+    // selected value. Don't show anything to prevent error that an option is
+    // missing
+    if (!selectedOption && selected) {
+        return []
+    }
+
+    const optionsContainSelected = options?.find(
+        ({ value }) => value === selected
+    )
+
+    if (selectedOption && !optionsContainSelected) {
+        return [...options, selectedOption as SelectOption]
```

Comment:
Still very minor.
```ts
selectedOption as SelectOption
```
is still not needed, right? It's the `SelectedOption` type in the param, and it can only be `undefined` - but since you're checking for a truthy value in the if - the type should be narrowed to `SelectOption`? It doesn't give me any errors.

Reply:
Ah yeah, I only removed one type cast, removed the second one as well